File: backend/src/Ctor.ReportEngine/ReportEngine/src/main/kotlin/com/ctor/reportEngine/db/entities/Material.kt
package com.ctor.reportEngine.db.entities

import java.util.Date
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne

@Entity(name = "Material")
class Material(
    @Id var Id: Long,
    var MaterialTypeId: Long,
    var MeasurementId: Long,
    var Amount: Int,
    var CompanyId: Long,
    var Price: Double,
    var Date: Date,
    var CompanyAddress: String,
    var CompanyName: String
) {
    @ManyToOne
    @JoinColumn(name = "MaterialTypeId", insertable = false, updatable = false)
    lateinit var MaterialType: MaterialType

    @ManyToOne
    @JoinColumn(name = "MeasurementId", insertable = false, updatable = false)
    lateinit var Measurement: Measurement

}

File: backend/src/Ctor.Domain/Entities/Material.cs
using Ctor.Domain.Common;

namespace Ctor.Domain.Entities;
public class Material: BaseEntity
{
    public long? MaterialTypeId { get; set; }
    public MaterialType? MaterialType { get; set; }
    public string CompanyName { get; set; } = string.Empty;
    public string CompanyAddress { get; set; } = string.Empty;
    public long? MeasurementId { get; set; }
    public Measurement? Measurement { get; set; }
    public decimal Price { get; set; }
    public int Amount { get; set; }
    public DateTime Date { get; set; }

    public long? CompanyId { get; set; }
    public Company Company { get; set; }
    
    public ICollection<RequiredMaterial> RequiredMaterials { get; set; }
}
